aderyn_core/src/detect/detector.rs

```diff
@@ -489,4 +503,8 @@ pub trait IssueDetector: Send + Sync + 'static {
     fn hints(&self) -> BTreeMap<(String, usize, String), String> {
         BTreeMap::new()
     }
+
+    fn merge_conflict_resolution_strategy(&self) -> MergeConflictResolutionStrategy {
```

Comment:
Could you explain in more detail why this is needed @TilakMaddy ?

Is this something we'll need to add to other detectors?

Reply:
Yeah, so this depends on the nature of the detector. By default all detectors implement the Union Strategy however, this one should implement Intersection Strategy.

The idea behind this is that, say you have 2 files:
File 1: pragma solidity 0.8.20
(Think ChildContract)

File 2: pragma solidity ^0.8.0
(Think parent contract)

And code in file1 imports some stuff from file2, we will have the following contexts:

Context 1:
File1, File2

Context2:
File2

Notice that File2 appears in both contexts (we already knew this from ICF so not going in detail here)

The important part is that when a detector finds that some variable could be marked immutable just by looking at File2 in Context2 that doesn't make it safe to conclude that the variable could in fact be marked immutable.
The reason is that when the same detector looks at File2 from Context1, it will see that nothing can be marked immutable. 

Therefore, for the same file, for the same detector we have 2 different answers from 2 different contexts.
This is a case of merge conflict. So far, what we have done is reported the union of all issue instances. (Which is still the default strategy).

However here in this detector, we can only safely conclude that a variable could be marked immutable if for a given file, all the contexts that involve the said file, report the same. (We must take intersection here)
Therefore due to the nature of this detector it is given Intersection resolution strategy

@alexroan